gems/pending/VolumeManager/LVM/parser.rb

```diff
@@ -0,0 +1,140 @@
+require 'active_support/core_ext/object/try'
+
+class Lvm2MdParser
+  HASH_START      = '{'
+  HASH_END        = '}'
+  ARRAY_START     = '['
+  ARRAY_END       = ']'
+  STRING_START    = '"'
+  STRING_END      = '"'
+
+  attr_reader :vgName
+
+  def initialize(mdStr, pvHdrs)
+    @pvHdrs = pvHdrs        # PV headers hashed by UUID
+    @mda = mdStr.gsub(/#.*$/, "").gsub("[", "[ ").gsub("]", " ]").gsub('"', ' " ').delete("=,").gsub(/\s+/, " ").split(' ')
+    @vgName = @mda.shift
+  end # def initialize
+
+  def parse
+    vgHash = {}
+    parseObj(vgHash, @vgName)
+    vg = vgHash[@vgName]
+
+    getVgObj(@vgName, vg)
+  end # def parse
+
+  def self.dumpMetadata(md)
+    level = 0
+    md.lines do |line|
+      line.strip!
+      level -= 1 if line[0, 1] == HASH_END || line[0, 1] == ARRAY_END
+      $log.info((level > 0 ? "    " * level : "") + line)
```

Comment:
@movitto The tests are failing. When you change this to `$log.debug` you probably have to rebase your branch to pick up the fix for the test failures: https://github.com/ManageIQ/manageiq/pull/7909